src/dmd/escape.d

```diff
@@ -992,7 +992,7 @@ private void inferReturn(FuncDeclaration fd, VarDeclaration v)
     else
     {
         // Perform 'return' inference on parameter
-        if (tf.ty == Tfunction && tf.parameters)
+        if (tf.ty == Tfunction)
```

Comment:
This doesn't look like a refactoring change

Reply:
It's covered by line 997, where dim will be 0 for !tf.parameters, and the loop will execute 0 times.

Reply:
OK